### ISSUE TITLE
Treat the `<pre>` element as a container element

### ISSUE THREAD

**Issue by SaschaCowley** [nvaccess/nvda#17045]
### Is your feature request related to a problem? Please describe.

When browsing websites with code listings, it is often desirable to skip over code to read the surrounding context. Sighted users have the ability to do this as the code is typically formatted in a monospace font, and typically with a different colour scheme to the literary copy.

### Describe the solution you'd like

Treat the `<pre>` element (block-level pre-formatted text) as a container element (much like a list, table etc), thus enabling the `,` key to be used to jump to the end of the element.

### Describe alternatives you've considered

A "jump to next different formatting" command may work, though this is unlikely given many pages will have syntax highlighting that will be detected as different formatting.

Encourage web authors to put skip links above their code listings, though this would be very slow to become widespread (some sites do this), and would disadvantage keyboard but non-screen reader users.

### Additional context

The `<pre>` element is not exclusively for code listings. Other use cases include ASCII art, terminal output, and poetry listings. The `<code>` element is an inline element, so it would not be appropriate to treat it as a container. If limiting the scope of this command were desired, we could ensure it is only applied to `<pre>` elements that contain `<code>` elements as descendants.

Another issue is how to communicate the container type to users. Calling these containers "pre" would likely be confusing for the majority of users. Expanding it to "pre-formatted text" would likely also be confusing, as users would probably be confused because the rest of the website is formatted. Stating they are "monospaced text" is confusing semantics and styling, and may confuse users who then expect all monospace text to be so indicated. I think "code" would likely be the best solution, even if it doesn't fully encapsulate what the `<pre>` element is about. Or perhaps "plain text" to be more generic (is this a widely understood term?).

**Comment by CyrilleB79** [comment]
Very nice feature request!

I'd prefer "plain text", but I'm fine with "code".
In both cases, these terms are not self explaining so this should be well documented.

I'd even prefer "plain text block" or "code block" to denote the fact that it is a container, and thus that the jump to end command can be used.

**Comment by XLTechie** [comment]
I agree about the value of treating pre as a container, but am equally concerned about having to announce this somehow.

A thing to remember about pre, is that you can also just style a div with one of the white-space pre variations.
So we also need to catch styled containers that aren't pres.

As a partial solution, the previous/next style gestures might help, though as you pointed out, they will likely hiccup on highlighting.

I definitely support some version of this idea though, no matter how it is announced.

Would "Code block" be viable as a shorter title?
